api/client/proto/authservice.proto

```diff
@@ -946,13 +908,13 @@ service AuthService {
     rpc GenerateDatabaseCert(DatabaseCertRequest) returns (DatabaseCertResponse);
 
     // GetRole retrieves a role described by the given request.
-    rpc GetRole(GetRoleRequest) returns (types.RoleV3);
+    rpc GetRole(types.GetResourceRequest) returns (types.RoleV3);
     // GetRole retrieves all roles.
-    rpc GetRoles(google.protobuf.Empty) returns (GetRolesResponse);
+    rpc GetRoles(google.protobuf.Empty) returns (stream types.RoleV3);
```

Comment:
Why use a stream here?
For users or nodes it might make sense, since there are many. Roles will be O(100) at most.
Streaming gRPC is generally harder to use and implement, so I'd avoid it.

Reply:
To be honest I did this to avoid making `message Get****Response` for each resource. Didn't think of the downsides, will change it.